test/hotspot/jtreg/compiler/debug/TestStressDistinctSeed.java

```diff
@@ -0,0 +1,116 @@
+import jdk.test.lib.Asserts;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Arrays;
+
+/*
+ * @test
+ * @key stress randomness
+ * @requires vm.debug == true & vm.compiler2.enabled
+ * @requires vm.flagless
+ * @summary Tests that stress compilations with the N different seed yield different
+ *          IGVN, CCP, macro elimination, and macro expansion traces.
+ * @library /test/lib /
+ * @run driver compiler.debug.TestStressDistinctSeed
+ */
+
+public class TestStressDistinctSeed {
+
+    private static int counter = 0;
+
+    static String phaseTrace(String stressOption, String traceOption,
+            int stressSeed) throws Exception {
+        String className = TestStressDistinctSeed.class.getName();
+        String[] procArgs = {
+                "-Xcomp", "-XX:-TieredCompilation", "-XX:-Inline", "-XX:+CICountNative",
+                "-XX:CompileOnly=" + className + "::sum", "-XX:" + traceOption,
+                "-XX:+" + stressOption, "-XX:StressSeed=" + stressSeed,
+                className, "10" };
+        ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(procArgs);
+        OutputAnalyzer out = new OutputAnalyzer(pb.start());
+        out.shouldHaveExitValue(0);
+        return out.getStdout();
+    }
+
+    static String igvnTrace(int stressSeed) throws Exception {
+        return phaseTrace("StressIGVN", "+TraceIterativeGVN", stressSeed);
+    }
+
+    static String ccpTrace(int stressSeed) throws Exception {
+        return phaseTrace("StressCCP", "+TracePhaseCCP", stressSeed);
+    }
+
+    static String macroExpansionTrace(int stressSeed) throws Exception {
+        return phaseTrace("StressMacroExpansion",
+                "CompileCommand=PrintIdealPhase,*::*,AFTER_MACRO_EXPANSION_STEP",
+                stressSeed);
+    }
+
+    static String macroEliminationTrace(int stressSeed) throws Exception {
+        return phaseTrace("StressMacroElimination",
+                "CompileCommand=PrintIdealPhase,*::*,AFTER_MACRO_ELIMINATION_STEP",
+                stressSeed);
+    }
+
+    static void sum(int n) {
+        int[] arr1 = new int[n];
+        for (int i = 0; i < n; i++) {
+            synchronized (TestStressDistinctSeed.class) {
+                counter += i;
+                arr1[i] = counter;
+            }
+        }
+        System.out.println(counter);
+    }
+
+    public static void main(String[] args) throws Exception {
+        Set<String> igvnTraceSet = new HashSet<>();
+        Set<String> ccpTraceSet = new HashSet<>();
+        Set<String> macroExpansionTraceSet = new HashSet<>();
+        Set<String> macroEliminationTraceSet = new HashSet<>();
+        if (args.length == 0) {
+            for (int s = 0; s < 10; s++) {
+                igvnTraceSet.add(igvnTrace(s));
+                ccpTraceSet.add(ccpTrace(s));
+                macroExpansionTraceSet.add(macroExpansionTrace(s));
+                macroEliminationTraceSet.add(macroEliminationTrace(s));
```

Comment:
A suggestion, do you also want to check here that two runs with the same seed produce the same result to show that different seeds really produce different results due to the seed and not just some indeterminism with the test itself? How long does your test need now and afterwards with a fastdebug build? Maybe we can also lower the number of seeds if it takes too long or only do the equality-test for a single seed.

Reply:
Thanks for the update and the numbers! I agree that `N=5` seems reasonable. Looks good!